ios/RNTPTDocumentView.m

```diff
@@ -4604,6 +4619,45 @@ - (NSString *) getDocumentPath {
     return self.currentDocumentViewController.coordinatedDocument.fileURL.path;
 }
 
+#pragma mark - Get All Fields
+
+- (NSMutableArray<NSDictionary *> *)getAllFieldsForDocumentViewTag:(int)pageNumber
+{
+    PTPDFViewCtrl *pdfViewCtrl = self.currentDocumentViewController.pdfViewCtrl;
+    if (!pdfViewCtrl) {
+        return nil;
+    }
+    PTPDFDoc * doc = [pdfViewCtrl GetDoc];
+    NSMutableArray<NSDictionary *> *resultMap = [[NSMutableArray alloc] init];
+    NSError *error;
+    if(doc){
+        [pdfViewCtrl DocLockReadWithBlock:^(PTPDFDoc * _Nullable doc) {
+            PTPage *page = [doc GetPage:pageNumber];
+            int num_annots = [page GetNumAnnots];
+            for (int i = 0; i < num_annots; i ++){
+                PTAnnot *annot = [page GetAnnot:i];
+                if(annot != nil) {
+                    if ([annot GetType] == e_ptWidget) {
+                        __block NSMutableDictionary <NSString *, NSObject *> *fieldMap = [[NSMutableDictionary alloc] init];
+
+                        fieldMap = [self getFieldWithHasAppearance:annot];
+
+                        [resultMap addObject:fieldMap];
+                    }
+                }
+            }
+        } error:&error];
+        
+        if (error) {
+            NSLog(@"An error occurred: %@", error);
+            return nil;
+        }
+        
+        return resultMap; 
+    }
+    return nil;
+}
```

Comment:
I think we can get rid of the check for `doc != nil` — it would be handled with an error in the read lock block. It's also a little confusing as the `doc` inside the read block has the same variable name.

```suggestion
#pragma mark - Get All Fields

- (NSArray<NSDictionary *> *)getAllFieldsForDocumentViewTag:(int)pageNumber
{
    PTPDFViewCtrl *pdfViewCtrl = self.currentDocumentViewController.pdfViewCtrl;
    if (!pdfViewCtrl) {
        return nil;
    }
    NSMutableArray<NSDictionary *> *resultMap = [[NSMutableArray alloc] init];
    NSError *error;
    [pdfViewCtrl DocLockReadWithBlock:^(PTPDFDoc * _Nullable doc) {
        PTPage *page = [doc GetPage:pageNumber];
        int num_annots = [page GetNumAnnots];
        for (int i = 0; i < num_annots; i ++){
            PTAnnot *annot = [page GetAnnot:i];
            if(annot != nil) {
                if ([annot GetType] == e_ptWidget) {
                    __block NSMutableDictionary <NSString *, NSObject *> *fieldMap = [[NSMutableDictionary alloc] init];

                    fieldMap = [self getFieldWithHasAppearance:annot];

                    [resultMap addObject:fieldMap];
                }
            }
        }
    } error:&error];
        
    if (error) {
        NSLog(@"An error occurred: %@", error);
        return nil;
    }
        
    return [resultMap copy]; 
}
```